File: rust/bin/keygen.rs
// Part of liblithium, under the Apache License v2.0.
// SPDX-License-Identifier: Apache-2.0

use std::{
    env,
    fs::File,
    io::{Result, Write},
    process::exit,
};

use liblithium::sign::keygen;

fn main() -> Result<()> {
    let mut args = env::args();
    let prog = args.next().unwrap();
    let sk_path = args.next().unwrap_or_else(|| {
        eprintln!("usage: {} <key-filename>", prog);
        exit(1)
    });
    let (pk, sk) = keygen();
    File::create(&sk_path)?.write_all(&sk)?;
    File::create(sk_path + ".pub")?.write_all(&pk)
}

File: py/entry_points/keygen.py
# Part of liblithium, under the Apache License v2.0.
# SPDX-License-Identifier: Apache-2.0

from lithium.sign import keygen

import sys
from pathlib import Path


def main():
    if len(sys.argv) < 2:
        print(f"usage: {sys.argv[0]} <key-filename>", file=sys.stderr)
        sys.exit(1)

    pk, sk = keygen()

    Path(sys.argv[1]).write_bytes(sk)
    Path(sys.argv[1] + ".pub").write_bytes(pk)
